File: Python/A/988A-diverseteam.py
class Solver:
    def __init__(self):
        self.n = 0
        self.k = 0
        self.ratings = []

    def input_data(self):
        self.n, self.k = map(int, input().split())
        self.ratings = list(map(int, input().split()))

    def answer(self):
        unique = set(self.ratings)
        if len(unique) < self.k:
            print("NO")
        else:
            print("YES")
            for value in list(unique)[: self.k]:
                print(self.ratings.index(value) + 1, end=" ")


solution = Solver()
solution.input_data()
solution.answer()

File: C++/A/988A-diverseteam.cpp
#include <bits/stdc++.h>
using namespace std;

class Solver
{
    int n, k;
    vector<int> ratings;

public:
    void input()
    {
        cin >> n >> k;
        for (int i = 0; i < n; i++)
        {
            int x;
            cin >> x;
            ratings.push_back(x);
        }
    }
    void answer()
    {
        set<int> unique;
        for (int i = 0; i < n; i++)
            unique.insert(ratings[i]);
        if (unique.size() < k)
            cout << "NO";
        else
        {
            cout << "YES\n";
            for (auto it = unique.begin(); it != unique.end() && k--; it++)
                cout << distance(ratings.begin(), find(ratings.begin(), ratings.end(), *it)) + 1 << ' ';
        }
    }
};

int main()
{
    Solver solution;
    solution.input();
    solution.answer();
}